backend/routes/orders.js

```diff
@@ -0,0 +1,106 @@
+// routes/orders.js
+import express from 'express';
+import Stripe from 'stripe';
+import { Order } from '../models/Order.js';
+import { authenticateUser } from '../middleware/auth.js'; 
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const router = express.Router();
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY);
+
+// POST /orders/create-payment - Create payment intent and order
+router.post("/create-payment-intent", authenticateUser, async (req, res) => {
+  try {
+    const { items, shippingAddress } = req.body;
+    
+    // Calculate total with shipping
+    const totalAmount = items.reduce((sum, item) => sum + (item.price * item.quantity), 0);
+    const shippingCost = 49;
+    const finalAmount = totalAmount + shippingCost;
+
```

Comment:
here I would check if the quanitity is 0, to remove it from the cart or at least de-activate the Checkout button. Either take care of this here and/or in the frontend. 
Screenshot for reference:
<img width="692" alt="image" src="https://github.com/user-attachments/assets/de9490d5-5658-4994-a9cd-8a69fcb0f439" />